### PR TITLE
ci: Setup for v2 branch

### PR DESCRIPTION
**Pull request by pgrzesik**
Reported internally, needed in order to publish new versions of v2 branch if needed

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -18,9 +18,7 @@ jobs:
       - name: Retrieve dependencies from cache
         uses: actions/cache@v2
         with:
-          path: |
-            ~/.npm
-            node_modules
```
Comment (medikoo) [inline .github/workflows/publish.yml:23]: Why this changes?

Comment (pgrzesik) [reply .github/workflows/publish.yml]: it shouldn't be changed, fixed

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -30,7 +30,6 @@ jobs:
           registry-url: https://registry.npmjs.org
 
       - name: Install dependencies
-        if: steps.cacheNpm.outputs.cache-hit != 'true'
```
Comment (medikoo) [inline .github/workflows/publish.yml:33]: This I think should also stay intact

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -68,19 +67,15 @@ jobs:
...
         uses: actions/setup-node@v1
         with:
           node-version: 16.x
           registry-url: https://registry.npmjs.org
 
-      # Normally we have a guarantee that deps are already there, still it may not be the case when:
-      # - `master` build for same commit failed (and we still pushed tag manually)
-      # - We've pushed tag manually before `master` build finalized
       - name: Install dependencies
-        if: steps.cacheNpm.outputs.cache-hit != 'true'
```
Comment (medikoo) [inline .github/workflows/publish.yml:83]: Same here?

Comment (pgrzesik) [reply .github/workflows/publish.yml:83]: fixed

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -68,17 +68,14 @@ jobs:
...
         uses: actions/setup-node@v1
         with:
           node-version: 16.x
           registry-url: https://registry.npmjs.org
 
-      # Normally we have a guarantee that deps are already there, still it may not be the case when:
-      # - `master` build for same commit failed (and we still pushed tag manually)
-      # - We've pushed tag manually before `master` build finalized
       - name: Install dependencies
         if: steps.cacheNpm.outputs.cache-hit != 'true'
```
Comment (medikoo) [inline .github/workflows/publish.yml:80]: I've just spotted that we miss `id: cacheNpm` on `Retrieve dependencies from cache` step (without that this condition is ineffective)

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -126,16 +123,19 @@ jobs:
...
       - name: Install Node.js and npm
         uses: actions/setup-node@v1
         with:
           node-version: 16.x
           registry-url: https://registry.npmjs.org
 
-      # Note: No need to install dependencies as we have retrieved cached `node_modules` for very
-      #       same `package.json` as stored with recent `master `build
+      - name: Install dependencies
+        if: steps.cacheNpm.outputs.cache-hit != 'true'
```
Comment (medikoo) [inline .github/workflows/publish.yml:135]: We need to ensure `id: cacheNpm` on `Retrieve dependencies from cache`, for that to work

Comment (pgrzesik) [reply .github/workflows/publish.yml:138]: great catch

**Review by medikoo — APPROVED** [review]
Looks great 👍
